akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.internal
+
+import java.util.Optional
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter._
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.ShardRegion.StartEntity
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import akka.cluster.sharding.typed.ClusterShardingSettings.StateStoreModeDData
+import akka.cluster.sharding.typed.ShardingEnvelope
+import akka.cluster.sharding.typed.ShardingMessageExtractor
+import akka.cluster.sharding.typed.scaladsl
+import akka.cluster.sharding.typed.javadsl
+import akka.cluster.sharding.typed.ShardedDaemonProcessSettings
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+import akka.cluster.sharding.typed.scaladsl.Entity
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+import akka.japi.function
+import scala.compat.java8.OptionConverters._
+
+import scala.concurrent.duration.Duration
+import scala.reflect.ClassTag
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object ShardedDaemonProcessImpl {
+
+  object KeepAlivePinger {
+    sealed trait Event
+    case object Tick extends Event
+
+    def apply[T](
+        settings: ShardedDaemonProcessSettings,
+        identities: Set[EntityId],
+        shardingRef: ActorRef[ShardingEnvelope[T]]): Behavior[Event] =
+      Behaviors.setup { context =>
+        Behaviors.withTimers { timers =>
+          def triggerStartAll(): Unit = {
+            identities.foreach(id => shardingRef.toClassic ! StartEntity(id))
+          }
+
+          context.log.debug(
+            s"Starting Sharded Daemon Set KeepAlivePinger with ping interval ${settings.keepAliveInterval}")
+          timers.startTimerWithFixedDelay(Tick, settings.keepAliveInterval)
+          triggerStartAll()
+
+          Behaviors.receiveMessage {
+            case Tick =>
+              triggerStartAll()
+              context.log.debug("Periodic ping sent to [{}] processes", identities.size)
+              Behaviors.same
+          }
+        }
+      }
+  }
+
+  def idAndNameToEntityId(id: Int, name: String): EntityId = s"$name-$id"
+
+  def idFromEntityId(entityId: EntityId, name: String): Int =
+    stringIdFromEntityId(entityId, name).toInt
+
+  def stringIdFromEntityId(entityId: EntityId, name: String): String =
+    entityId.drop(name.length + 1)
+
+  final class MessageExtractor[T](name: String) extends ShardingMessageExtractor[ShardingEnvelope[T], T] {
+    def entityId(message: ShardingEnvelope[T]): String = message match {
+      case ShardingEnvelope(id, _) => id
+    }
+
+    def shardId(entityId: String): String = stringIdFromEntityId(entityId, name)
```

Comment:
why do we have to include the `name` in the entityId? can't the entityId and shardId just be toString of the numeric index? Different names are separated by different entityTypeKey.

Reply:
Can't se a reason why we have to, I'll remove